### PR TITLE
Decompile DRE func_801A2848

### PR DESCRIPTION
**Pull request by bismurphy**
Kind of a messy function to match, but it matches. Looks like there are a few functions in this file with similar structure.

### CODE REVIEW

File: src/st/dre/22848.c [hunk 1]
```diff
@@ -9,8 +9,61 @@ long RotTransPers3(SVECTOR* v0, SVECTOR* v1, SVECTOR* v2, long* sxy0,
...
-Primitive* func_801A2848(Primitive* prim, u8* model, u16* arg2);
-INCLUDE_ASM("st/dre/nonmatchings/22848", func_801A2848);
+Primitive* func_801A2848(Primitive* prim, u8* arg1, u16* arg2) {
+    s32 pad0;
+    s32 pad1;
+    s32 pad2;
+    s32 pad3;
+    s32 pad4;
+
+    pad0 = *SPAD(arg1[0]);
```
Comment (Xeeynamo) [inline src/st/dre/22848.c:19]: Wanna use `SP` instead?

**Review by Xeeynamo — APPROVED** [review]
LGTM, just left a minor comment

File: src/st/dre/22848.c (hunk 1 above)
Comment (Xeeynamo) [reply src/st/dre/22848.c:19]: What if SPAD internally uses SP? Reason is the scratch pad offset on PSP is different and on PC I got far too many unexpected crashes not knowing a hardcoded address was used for the scratchpad. Using SP, even if SPAD is wrapping it, will at least a compilation error

Comment (bismurphy) [reply src/st/dre/22848.c:19]: Sure, that sounds like a nice idea. Will implement.

**Pull request merged**